Src/DependencyCollector/Shared/ServiceBusDiagnosticsListener.cs

```diff
@@ -0,0 +1,172 @@
+        {
+            this.telemetryClient = new TelemetryClient(configuration);
+        }
+
+        /// <summary>
+        /// Gets custom handler <see cref="HandleDiagnosticsEvent{TContext}"/>delegate.
+        /// </summary>
+        internal HandleDiagnosticsEvent<HashSet<string>> EventHandler => this.HandleEvent;
+
+        private void HandleEvent(KeyValuePair<string, object> evnt, DiagnosticListener diagnosticListener, HashSet<string> context)
+        {
+            // there is no guarantee IsEnabled was called in case there are multiple listeners in the system. i.e. we need to check it again
+            if (evnt.Key.EndsWith("Start", StringComparison.Ordinal))
+            {
+                return;
+            }
+
+            Activity currentActivity = Activity.Current;
+
+            switch (evnt.Key)
+            {
+                case "Microsoft.Azure.ServiceBus.ProcessSession.Stop":
+                case "Microsoft.Azure.ServiceBus.Process.Stop":
+                    this.OnRequest(evnt.Key, evnt.Value, currentActivity);
+                    break;
+                case "Microsoft.Azure.ServiceBus.Exception":
+                    this.OnException(evnt.Key, evnt.Value);
+                    break;
+                default:
+                    this.OnDependency(evnt.Key, evnt.Value, currentActivity);
+                    break;
+            }
+        }
+
+        private void OnDependency(string name, object payload, Activity activity)
+        {
+            DependencyTelemetry telemetry = new DependencyTelemetry { Type = DependencyType };
+
+            this.SetCommonProperties(name, payload, activity, telemetry);
+
+            // Endpoint is URL of particular ServiceBus, e.g. sb://myservicebus.servicebus.windows.net/
+            telemetry.Data = this.FetchPayloadProperty<Uri>(name, EndpointPropertyName, payload)?.ToString();
+
+            // Queue/Topic name, e.g. myqueue/mytopic
+            telemetry.Target = this.FetchPayloadProperty<string>(name, EntityPropertyName, payload);
+
+            this.telemetryClient.TrackDependency(telemetry);
+        }
+
+        private void OnRequest(string name, object payload, Activity activity)
+        {
+            RequestTelemetry telemetry = new RequestTelemetry();
+
+            this.SetCommonProperties(name, payload, activity, telemetry);
+
+            // Endpoint is URL of particular ServiceBus, e.g. sb://myservicebus.servicebus.windows.net/
+            telemetry.Url = this.FetchPayloadProperty<Uri>(name, EndpointPropertyName, payload);
+
+            // Entity 
+            // We want to make Source field extendable at the beginning as we may add
+            // multi ikey support and also build some special UI for requests coming from the queues
+            // so the Source looks like roleName:queueName (using the multi ikey schema).
+            telemetry.Source = "roleName:" + this.FetchPayloadProperty<string>(name, EntityPropertyName, payload);
```

Comment:
like with ```Dependency.Target``` - this is not unique enough (i.e. what if 2 SB accounts have a queue with the same name?)

Reply:
See above.